packages/client/src/devices/devices.ts

```diff
@@ -16,49 +19,28 @@ import { getLogger } from '../logger';
  * @param constraints the constraints to use when requesting the devices.
  * @param kind the kind of devices to enumerate.
  */
-const getDevices = (
-  constraints: MediaStreamConstraints,
-  kind: MediaDeviceKind,
-) => {
-  return new Observable<MediaDeviceInfo[]>((subscriber) => {
-    const enumerate = async () => {
+const getDevices = (permission: BrowserPermission, kind: MediaDeviceKind) => {
+  return from(
+    (async () => {
       let devices = await navigator.mediaDevices.enumerateDevices();
-      // some browsers report empty device labels (Firefox).
-      // in that case, we need to request permissions (via getUserMedia)
-      // to be able to get the device labels
-      const needsGetUserMedia = devices.some(
+      // for privacy reasons, most browsers don't give you device labels
+      // unless you have a corresponding camera or microphone permission
+      const shouldPromptForBrowserPermission = devices.some(
         (device) => device.kind === kind && device.label === '',
       );
-      if (needsGetUserMedia) {
-        let mediaStream: MediaStream | undefined;
-        try {
-          mediaStream = await navigator.mediaDevices.getUserMedia(constraints);
-          devices = await navigator.mediaDevices.enumerateDevices();
-        } finally {
-          if (mediaStream) disposeOfMediaStream(mediaStream);
-        }
+      if (shouldPromptForBrowserPermission) {
+        await permission.prompt({ throwOnNotAllowed: true });
```

Comment:
Enumerating devices is not an important enough action to allow re-prompting for a permission, so we use the default here (`forcePrompt: false`).